rewrite-maven/src/main/java/org/openrewrite/maven/tree/ResolvedPom.java

```diff
@@ -343,7 +343,9 @@ class Resolver {
         MavenPomDownloader downloader;
 
         public ResolvedPom resolve() throws MavenDownloadingException {
-            resolveParentsRecursively(requested);
+            if (!MavenExecutionContextView.view(ctx).getSkipDependencyResolution()) {
+                resolveParentsRecursively(requested);
```

Comment:
To detect Java versions in the CLI it'd be helpful to still resolve parent poms without resolving dependencies. That aligns with how we've seen customers use archetypes to set the Java version.  Would it make sense to make resolving parent poms a separate option?

Reply:
I'm not liking the options if that's the case. Almost would rather a general purpose dependency resolution interceptor pattern than an array of very specific options.